pkg/cli/cvd.go

```diff
@@ -350,20 +351,30 @@ func (c *cvdCreator) createCVDFromLocalSrcs() ([]*hoapi.CVD, error) {
 	return res.CVDs, nil
 }
 
-func credentialsFactoryFromSource(source string) (CredentialsFactory, error) {
+func credentialsFactoryFromSource(source string, projectID string) (CredentialsFactory, error) {
 	switch source {
 	case NoneCredentialsSource:
-		return func() string { return "" }, nil
+		return func() client.BuildAPICredential {
+			return client.BuildAPICredential{AccessToken: "", UserProjectID: ""}
+		}, nil
 	case InjectedCredentialsSource:
-		return func() string { return client.InjectedCredentials }, nil
+		return func() client.BuildAPICredential {
+			return client.BuildAPICredential{AccessToken: client.InjectedCredentials, UserProjectID: projectID}
```

Comment:
if using `client.InjectedCredentials` the CO is the one handling the access token, hence the CO should be responsible to set the UserProjectID value as well. If using "injected" we should return an error if the client sets a project a id.